app/models/targeting.rb

```diff
@@ -0,0 +1,41 @@
+class Targeting < ActiveRecord::Base
+
+  STATIC_TYPE = 'static_query'
+  DYNAMIC_TYPE = 'dynamic_bookmark'
+  TYPES = {STATIC_TYPE => N_('Static Query'), DYNAMIC_TYPE => N_('Dynamic Bookmark')}
+
+  belongs_to :user
+  belongs_to :bookmark
+
+  has_many :targeting_hosts, :dependent => :destroy
+  has_many :hosts, :through => :targeting_hosts
+  has_many :template_invocations, :dependent => :destroy
+
+  validates :targeting_type, :presence => true, :inclusion => Targeting::TYPES.keys
+  validates :bookmark, :presence => true, :if => "dynamic?"
+
+  before_create :assign_search_query, :if => :static?
```

Comment:
I think there are a few use cases here, lets spell them out to make it clear:

A) I want to execute a command against a bookmark at some point in the future using the current query of the bookmark
B) I want to execute a command against a bookmark at some point in the future using the future query of the bookmark
C) I want to execute a command against a bookmark at some point in the future using the hosts that the bookmark resolves right now

(I'm excluding execute NOW as that is more simple).    Which of these do we want to support?  With the current implementation only A & B are supported.   It sounds like @ares you may be suggesting we support C) ?